hugr-py/pyproject.toml

```diff
@@ -26,6 +26,8 @@ repository = "https://github.com/CQCL/hugr"
 [tool.poetry.dependencies]
 python = ">=3.10"
 pydantic = ">=2.7,<2.9"
+pydantic-extra-types = "^2.9.0"
+semver = "^3.0.2"
```

Comment:
nit: does this belong in this PR?

Reply:
Ah, you added Rust semver in an earlier PR but didn't update the python, and tests added here reveal this?

Reply:
the earlier PR added versions to the rust struct, but there was not matching python model. This PR adds the python model, and with it the semver dependency for the python model.